src/wallet/rpcwallet.cpp

```diff
@@ -2190,6 +2190,7 @@ UniValue getwalletinfo(const UniValue& params, bool fHelp)
             "  \"keypoolsize\": xxxx,        (numeric) how many new keys are pre-generated\n"
             "  \"unlocked_until\": ttt,      (numeric) the timestamp in seconds since epoch (midnight Jan 1 1970 GMT) that the wallet is unlocked for transfers, or 0 if the wallet is locked\n"
             "  \"paytxfee\": x.xxxx,         (numeric) the transaction fee configuration, set in " + CURRENCY_UNIT + "/kB\n"
+            "  \"masterkeyid\": \"<addr>\",    (string) the P2PKH address of the hd master key\n"
```

Comment:
The problem is, we store the `masterkeyid` as a `CKeyID`. IMO to get the xpub (base58 check encoded extended public key) we need to have a unlocked wallet.
@sipa: What would you propose instead? Caching the xpub in a separate wallet.dat record type?